pkg/resourceinterpreter/framework/interface.go

```diff
@@ -0,0 +1,39 @@
+package framework
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+)
+
+// Interpreter describe the interpretation rules
+type Interpreter interface {
```

Comment:
I don't think an interface could be called a `framework`. It'd be better to put this along with `ResourceInterpreter interface`.